espnet/bin/asr_recog.py

```diff
@@ -79,11 +80,22 @@ def main(args):
                         help='Word RNNLM model config file to read')
     parser.add_argument('--word-dict', type=str, default=None,
                         help='Word list to read')
-    parser.add_argument('--lm-weight', default=0.1, type=float,
-                        help='RNNLM weight.')
-    parser.add_argument('--streaming-window', type=int, default=False,
-                        help='Use streaming recognizer for inference - provide window size in frames. '
-                             '--batchsize must be set to 0 to enable this mode')
+    parser.add_argument('--lm-weight', type=float, default=0.1,
+                        help='RNNLM weight')
+    # streaming related
+    parser.add_argument('--streaming-mode', type=str, default=None,
+                        choices=['window', 'segment'],
+                        help="""Use streaming recognizer for inference
+                        (0: none, 1: window streaming, 2: segment streaming).
+                        `--batchsize` must be set to 0 to enable this mode""")
+    parser.add_argument('--streaming-window', type=int, default=10,
+                        help='Window size')
+    parser.add_argument('--min-blank-dur', type=int, default=10,
```

Comment:
How about adding `streaming` in the prefix of the options (e.g., `--streaming-min-blank-dur`, `--streaming-onset-margin`) so that we can easily understand that these options are related to streaming methods.